packages/flutter/lib/src/material/expansion_tile.dart

```diff
@@ -187,20 +201,30 @@ class _ExpansionTileState extends State<ExpansionTile> with SingleTickerProvider
 
   @override
   void didChangeDependencies() {
-    final ThemeData theme = Theme.of(context);
-    _borderColorTween
-      ..end = theme.dividerColor;
-    _headerColorTween
-      ..begin = theme.textTheme.subtitle1.color
-      ..end = theme.accentColor;
-    _iconColorTween
-      ..begin = theme.unselectedWidgetColor
-      ..end = theme.accentColor;
-    _backgroundColorTween
-      ..end = widget.backgroundColor;
+    _updateColorTweens();
     super.didChangeDependencies();
   }
 
+  @override
+  void didUpdateWidget(ExpansionTile oldWidget) {
+    _updateColorTweens();
+    super.didUpdateWidget(oldWidget);
+  }
+
+    void _updateColorTweens() {
+      final ThemeData theme = Theme.of(context);
+      _borderColorTween
+        ..end = theme.dividerColor;
+      _headerColorTween
+        ..begin = widget.headerColorTween?.begin ?? theme.textTheme.subtitle1.color
+        ..end = widget.headerColorTween?.end ?? theme.accentColor;
+      _iconColorTween
+        ..begin = widget.iconColorTween?.begin ?? theme.unselectedWidgetColor
+        ..end = widget.iconColorTween?.end ?? theme.accentColor;
+      _backgroundColorTween
+        ..end = widget.backgroundColor;
+    }
```

Comment:
Indentation here is off by two spaces

Reply:
Any tips on how to format the code properly and maybe automatically? I use VSCode and have Prettier installed, but whenever I save a file, it formats more than I want. So I turn it off and these kind of things pass :/

Reply:
I don't think there is a good way unless you implement your own formatting rules. I currently also just format code without any help from tools like Prettier